### PR TITLE
CancellationToken for Async methods

### PR DESCRIPTION
**Pull request by DanieleSky**
Add CancellationToken parameter for Async method and pass it to Dapper

### CODE REVIEW

**Comment by henkmollema** [comment]
Hi @DanieleSky, thanks for this PR! I have one comment: `CancellationToken` should be used a parameter rather than `CancellationTokenSource`. Could you change that?

**Comment by DanieleSky** [comment]
Hi @henkmollema , I've done!

File: src/Dommel/Dommel.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <Description>Simple CRUD operations for Dapper.</Description>
-    <PackageTags>dommel;crud;dapper;database;orm</PackageTags>
+    <PackageTags>para</PackageTags>
```
Comment (henkmollema) [inline src/Dommel/Dommel.csproj:4]: Can you undo this change?

File: src/Dommel/Get.cs [hunk 2]
```diff
@@ -33,13 +34,14 @@ public static partial class DommelMapper
         /// <param name="connection">The connection to the database. This can either be open or closed.</param>
         /// <param name="id">The id of the entity in the database.</param>
         /// <param name="transaction">Optional transaction for the command.</param>
+        ///  <param name="cancellationToken">Optional cancellationToken for the command.</param>
         /// <returns>The entity with the corresponding id.</returns>
-        public static async Task<TEntity?> GetAsync<TEntity>(this IDbConnection connection, object id, IDbTransaction? transaction = null)
+        public static async Task<TEntity?> GetAsync<TEntity>(this IDbConnection connection, object id, IDbTransaction? transaction = null, CancellationToken? cancellationToken = null)
```
Comment (henkmollema) [inline src/Dommel/Get.cs:39]: I think it's a best practice to use this pattern, rather than a nullable:

```suggestion
        public static async Task<TEntity?> GetAsync<TEntity>(this IDbConnection connection, object id, IDbTransaction? transaction = null, CancellationToken cancellationToken = default)
```

Can you apply this everywhere? You can also remove the `?? default` later then.

File: src/Dommel/Get.cs [hunk 3]
```diff
@@ -33,13 +34,16 @@ public static partial class DommelMapper
...
         /// <returns>The entity with the corresponding id.</returns>
-        public static async Task<TEntity?> GetAsync<TEntity>(this IDbConnection connection, object id, IDbTransaction? transaction = null)
+        public static async Task<TEntity?> GetAsync<TEntity>(this IDbConnection connection, object id, IDbTransaction? transaction = null, CancellationToken cancellationToken = default)
             where TEntity : class
         {
             var sql = BuildGetById(GetSqlBuilder(connection), typeof(TEntity), id, out var parameters);
             LogQuery<TEntity>(sql);
-            return await connection.QueryFirstOrDefaultAsync<TEntity>(sql, parameters, transaction);
+            //with CommandDefinition we can pass also the the cancellation token 
+            //for further informations https://github.com/DapperLib/Dapper/blob/main/Dapper/CommandDefinition.cs
```
Comment (henkmollema) [inline src/Dommel/Get.cs:45]: @DanieleSky you can remove these comments imo. The code is clear enough.

**Review by henkmollema — CHANGES_REQUESTED** [review]

File: src/Dommel/Get.cs (hunk 3 above)
Comment (DanieleSky) [reply src/Dommel/Get.cs:45]: Hi @henkmollema , you mean all comments like this or only the rows 44-45?

Comment (henkmollema) [reply src/Dommel/Get.cs:45]: I meant all the comments indeed.

Comment (DanieleSky) [reply src/Dommel/Get.cs]: ok, I've done

**Review by henkmollema — APPROVED** [review]
Great, thank you!
